middleware/authz.js

```diff
@@ -0,0 +1,22 @@
+require("express-async-errors");
+const jwt = require("jsonwebtoken");
+
+module.exports =  (listOfRolesAllowed) => {
+  return async (req, res, next) => {
+    const token = req.header("x-auth-token");
+    if (!token)
+      return res.status(401).send(`Access denied. No token provided.`);
+
+    const decoded = jwt.verify(token, process.env.JWT_PRIVATE_KEY);
+    const isRoleAllowed = listOfRolesAllowed.find((role) =>
+      role == decoded.isManager ? "Manager" : "Client"
```

Comment:
Use constants instead of hardcoding strings for roles and error messages. This improves maintainability by centralizing values that may change and helps avoid potential typos or inconsistencies in your code.